### DISCUSSION TITLE
How to open multiple selected files in vim or in other program?

### DISCUSSION THREAD

**Discussion by nivit** [kovidgoyal/kitty#6116] [Q&A]
Hi,

I have this in kitty.conf

`map kitty_mod+a>e kitten hints --type=path --add-trailing-space=never --multiple --program "launch --type=tab vim"`

but it only works if the current working directory is $HOME. In other directories vim creates new files (in $HOME). Adding the option `--cwd=oldest` or `--cwd=last_reported` to `launch`, only the first selected file is edited, the other ones are created ex-novo.

So what is the solution?

**Comment by kovidgoyal** [comment]
Works fine for me running

```
kitty -o 'map f1 kitten hints --type=path --add-trailing-space=never --multiple --program "launch --type=tab vim"'
```

Then in the kitty window
cd somedir
ls
press f1
select two files
press esc
the two files are opened in vim as expected

**Comment by nivit** [comment]
Thanks for the answer, but apparently it only works in Linux, in FreeBSD the behaviour is as described before. It seems that only the first selected file "receives" the correct working directory (if I pass the `--cwd` option, otherwise all files are created by scratch), the other ones receive the specificed directory when launching kitty (default $HOME).

**Reply by kovidgoyal** [reply]
That would imply kitty is failing to get the working dir of a process on freebsd, which IIRC is done using pwdx presumably that is missing or broken on your system

**Reply by nivit** [reply]
That's the reason! `pwdx` returned an empty string, because I had the kernel knob `security.bsd.unprivileged_proc_debug` set to `0`. Allowing non root-users to access to debugging facilities, `pwdx` now shows the current working directory of the process. Thanks a lot!
